nixos/modules/services/mail/rss2email.nix

```diff
@@ -0,0 +1,226 @@
+        font-size: 20px;
+        text-decoration: none;
+      }
+      .footer {
+        background: #c3d9ff;
+        border-top: solid 4px #c3d9ff;
+        padding: 5px;
+        margin-bottom: 0px;
+      }
+      #entry {
+        border: solid 4px #c3d9ff;
+      }
+      #body {
+        margin-left: 5px;
+        margin-right: 5px;
+      }
+    '';
+    unicode-snob = "False";
+    links-after-each-paragraph = "False";
+    body-width = "0";
+    email-protocol = "sendmail";
+    sendmail = "sendmail";
+    smtp-auth = "False";
+    smtp-username = "username";
+    smtp-password = "password";
+    smtp-server = "smtp.yourisp.net:25";
+    smtp-ssl = "False";
+    smtp-ssl-protocol = "SSLv3";
+    imap-auth = "False";
+    imap-username = "username";
+    imap-password = "password";
+    imap-server = "imap.yourisp.net";
+    imap-port = "143";
+    imap-ssl = "False";
+    imap-mailbox = "INBOX";
+    verbose = "warning";
+  };
+
+in {
+
+  ###### interface
+
+  options = {
+
+    services.rss2email = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable rss2email.";
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.rss2email;
+        defaultText = "pkgs.rss2email";
+        description = "The rss2email package to use.";
+      };
+
+      to = mkOption {
+        type = types.str;
+        description = "Mail address to which to send emails";
+      };
+
+      interval = mkOption {
+        type = types.str;
+        default = "1h";
+        description = "How often to check the feeds, in systemd interval format";
+      };
+
+      configuration = mkOption {
+        type = types.attrs;
+        description = ''
+          The configuration to give rss2email.
+
+          Default will use system-wide <literal>sendmail</literal> to send the
+          email. This is rss2email's default when running
+          <literal>r2e new</literal>.
+
+          This set contains key-value associations that will be set in the
+          <literal>[DEFAULT]</literal> block along with the
+          <literal>to</literal> parameter.
+        '';
+        example = lib.literalExample ''
+          {
+            from = "user@rss2email.invalid";
+
+            # You likely want to use something like this for css
+            css = builtins.replaceStrings ["\n"] [""] '''
+              body {
+                font: 15pt Arial;
+              }
+            ''';
+          }
+        '';
+        default = default-config;
+      };
+
+      feeds = mkOption {
+        description = "The feeds to watch.";
+        type = types.attrsOf (types.submodule {
+          options = {
+            url = mkOption {
+              type = types.str;
+              description = "The URL at which to fetch the feed.";
+            };
+
+            to = mkOption {
+              type = with types; nullOr str;
+              default = null;
```

Comment:
Along with the `config` change, you can then make these changes here:
- Change the type to `str`
- Remove the default value here
- Set the default by doing `to = mkDefault cfg.to;` in the `config` section of the submodule.

Reply:
This is possible indeed, but it moves the module farther away from upstream configuration, as upstream configuration just doesn't add a `to`, and I tried to replicate upstream behavior as well as possible… now you're right, there's little else that could legitimately be done other than using the system-wide `to`, so I guess this won't change visible behavior :)

Reply:
Ah, I think your current implementation always just defaults to `cfg.to` though (line 12). So if you want to replicate upstream behavior, you should probably just not set it there instead.

Reply:
I think [lines 11-12](https://github.com/NixOS/nixpkgs/blob/829c58e2dec794347bac63faabf6f3f8819dcf17/nixos/modules/services/mail/rss2email.nix#L11-L12) correctly set `url` from `feed` and add the `to` field iff. `cfg.to` is not `null`, thanks to the `optionalAttrs` (and testing validates this, if I'm not mistaken on the testing process).

I guess you confused with [line 9](https://github.com/NixOS/nixpkgs/blob/829c58e2dec794347bac63faabf6f3f8819dcf17/nixos/modules/services/mail/rss2email.nix#L9), where I did set `to` unconditionally, but in the `[DEFAULT]` block?